sorts/heapsort.ts

```diff
@@ -0,0 +1,46 @@
+/* 
+Implementation of in-place heap sort.
+
+  complexity:
+    best time - O(N log N)
+    avg time - O(N log N)
+    worst time - O(N log N)
+
+    memory - O(1)
+    in-place - Yes
+    stable - No
+*/
+
+module.exports = function HeapSort(arr: number[]): number[] {
+  const left = (i: number): number => i * 2 + 1,
+    right = (i: number): number => i * 2 + 2;
```

Comment:
missing tab? 🤔

Reply:
You mean it's not indented properly? That what my formatter do to my code 🙃